File: gradle/src/main/kotlin/hello/HelloWorld.kt
package hello

import org.apache.commons.lang3.time.DateFormatUtils

fun printGreeting(args: Array<String>): String {
    return if (args.isNotEmpty()) {
        val name = args.drop(0).joinToString(" ")
        "Hi $name, greetings from Gradle !"
    } else {
        "Hello World from Gradle !"
    }
}

fun printDate(): String {
    val currentDate = System.currentTimeMillis()
    val format = "MMMM dd, yyyy HH:mm"
    val formattedDate = DateFormatUtils.format(currentDate, format)
    return "Today is $formattedDate"
}

File: swiftpm/Sources/HelloWorld/HelloWorld.swift
import Foundation

public func printGreeting(_ args: [String]) -> String {
  if args.count > 1 {
    let name = args[1...].joined(separator: " ")
    return "Hi \(name), greetings from SwiftPM !"
  } else {
    return "Hello World from SwiftPM !"
  }
}

public func printDate() -> String {
  let dateFormatter = DateFormatter()
  dateFormatter.dateFormat = "MMMM d, yyyy 'at' h:mm a"
  let currentDate = dateFormatter.string(from: Date())
  return "Today is \(currentDate)"
}
